### PR TITLE
Add op uuid to image replication

### PR DESCRIPTION
**Pull request by cd1989**
Add uuid to track all jobs triggered by one replication execution.

Related issue: https://github.com/goharbor/harbor/issues/3876

### CODE REVIEW

**Comment by steven-zou** [comment]
@ywk253100 

Let's take a look at this PR.

File: src/core/api/replication.go [hunk 1]
```diff
@@ -78,16 +81,27 @@ func (r *ReplicationAPI) Post() {
...
+
+	r.Data["json"] = api_models.ReplicationResponse{
+		UUID: opUUID,
+	}
+	r.ServeJSON()
 }
 
-func startReplication(policyID int64) error {
-	return notifier.Publish(topic.StartReplicationTopic,
+// startReplication triggers a replication and return the uuid of this replication.
```
Comment (clouderati) [inline src/core/api/replication.go]: @reasonerjt Is there a reason that we don't log here that we've started a replication job (and in this case explicitly state the UUID of the job)?

Comment (reasonerjt) [reply src/core/api/replication.go]: One replication job is designed to handle replication of one repository,
and one replication action, given certain filter, will map to multiple jobs in job service.
And this PR make it possible to trace a job based on replication action.

**Comment by cd1989** [comment]
@ywk253100 Is it possible to make this into Harbor 1.7.0 ?

**Comment by ywk253100** [comment]
@cd1989 we'll discuss later, thanks.

**Comment by steven-zou** [comment]
@cd1989 

For the purpose of this PR, do you have any real scenario or customer requirements?

**Comment by cd1989** [comment]
@steven-zou Yes. In our product, we need to get replication results of one replication action. If users trigger a replication manually, they want to get the status of this replication action. We are not using Harbor independently, but call APIs of Harbor and embed Harbor to our product.  Also, we have customers that have similar requirements.

**Comment by clouderati** [comment]
I'm definitely in favor of this. Having a UUID that we can trace throughout various internal calls is tremendously invaluable.

**Comment by cd1989** [comment]
@steven-zou @clouderati What's the conclusion of this PR, do we accept it in 1.7.0 ?

**Comment by clouderati** [comment]
I'm ok with this landing in v1.7. @reasonerjt and @steven-zou – can we please make sure this gets merged if it's not breaking anything? Thanks.

**Comment by steven-zou** [comment]
@clouderati @cd1989 

I think it's ok to merge into master. I'll do it now.

**Comment by cd1989** [comment]
thx, @steven-zou

**Pull request merged**